### PR TITLE
Added Android sensors to android_core

### PR DESCRIPTION
**Pull request by chadrockey**
Add new sensors (code used to live in Android Sensors Driver).  Also updated Orientation publisher.

Since these are new files (and one minor change), should be fine to merge into hydro branch and not wait for indigo.  @damonkohler @stonier, if we think the changes to OrientationPublisher are too great, we can hold off on that one and merge the rest.

### CODE REVIEW

**Comment by chadrockey** [comment]
I added the rest of the obvious API today.  Some of these came from newer versions, so I added android_ice_cream_sandwich (14) and android_jelly_bean_mr2.

Looks like the Fused Location API is actually in... Google Play Services of all things.  That means it shouldn't belong here.

File: android_gingerbread_mr1/src/main/java/org/ros/android/IlluminancePublisher.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        {
+            this.ilThread.join();
+        }
+        catch (InterruptedException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    //@Override
```
Comment (damonkohler) [inline android_gingerbread_mr1/src/main/java/org/ros/android/IlluminancePublisher.java:186]: Why are these commented out?

File: android_gingerbread_mr1/src/main/java/org/ros/android/IlluminancePublisher.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    public void onShutdown(Node arg0)
+    {
+        if(this.ilThread == null){
+            return;
+        }
+
+        this.ilThread.shutdown();
+
+        try
```
Comment (damonkohler) [inline android_gingerbread_mr1/src/main/java/org/ros/android/IlluminancePublisher.java:176]: Please try to be consistent with style. Curly braces at the end of a line where the block starts, space before the curly brace, two space indents, don't use avoid abbreviations, don't use things like "arg0", etc.

**Comment by damonkohler** [comment]
I have a PR going out today that changes the naming scheme of the library projects to use the API level instead (e.g. android_10, android_15, etc.). It's more concise and consistent with Android tooling terminology.

File: android_gingerbread_mr1/src/main/java/org/ros/android/IlluminancePublisher.java (hunk 1 above)
Comment (chadrockey) [inline android_gingerbread_mr1/src/main/java/org/ros/android/IlluminancePublisher.java:186]: I think my IDE had an older Java version at some point, this appears to have been a helpful correction.  The Android Studio linter didn't complain, but I will make sure these are all correct.

http://stackoverflow.com/a/8737207

**Comment by chadrockey** [comment]
Great, once the new versions are in, I'll send this against master.

**Comment by tulku** [comment]
@chadrockey, now that the libraries changed names, can you rebase this PR to master?

**Comment by damonkohler** [comment]
Ping @chadrockey.

**Pull request closed without merging**